x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/IndexDeprecationChecks.java

```diff
@@ -258,6 +272,76 @@ static DeprecationIssue tooManyFieldsCheck(IndexMetaData indexMetaData) {
         return null;
     }
 
+    static DeprecationIssue deprecatedDateTimeFormat(IndexMetaData indexMetaData) {
+        List<String> fields = new ArrayList<>();
+
+        fieldLevelMappingIssue(indexMetaData, ((mappingMetaData, sourceAsMap) -> fields.addAll(
+            findInPropertiesRecursively(mappingMetaData.type(), sourceAsMap,
+                IndexDeprecationChecks::isDateFieldWithDeprecatedPattern
+                , IndexDeprecationChecks::formatDateField))));
+
+
+        if (fields.size() > 0) {
+            return new DeprecationIssue(DeprecationIssue.Level.WARNING,
+                "Date time field format likely contain deprecated pattern",
+                "https://www.elastic.co/guide/en/elasticsearch/reference/7.0/breaking-changes-7.0.html#breaking_70_java_time_changes",
+                "This index has date fields with deprecated formats: " + fields + "");
+        }
+        return null;
+    }
+
+    private static boolean isDateFieldWithDeprecatedPattern(Map<?, ?> property) {
+        return "date".equals(property.get("type")) &&
+                property.containsKey("format") &&
+            isDeprecatedFormat((String) property.get("format"));
+    }
+
+    static Map<String, String> JODA_PATTERNS_DEPRECATIONS = new LinkedHashMap<>();
+    static {
+        JODA_PATTERNS_DEPRECATIONS.put("Y", "'Y' year-of-era becomes 'y'. 'Y' means week-based-year.");
```

Comment:
We already have these deprecation messages in `Joda`, which are logged when the patterns are parsed. This seems to be doubling them? Additionally, I think some of these are too broad, like the check for `y`, which users wouldn't need to do anything about. Any deprecation messages should be actionable.

Reply:
Yes - that was just a draft, but next step is to have those messages in one place and use in both Joda & Deprecation API.
for `y` pattern check I think it would be fair to notify users about this as it changes to `u` (I missed this on previous commit). Some of the patterns are not actionable though - like `C` century is no longer present in java.time.